### PR TITLE
chore(website): consistent .mdx file naming

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## Overview

Spinning off from #6318:

* Makes all website Markdown files `Upper_Snake_Case.mdx`
    * As @Josh-Cena has had to tell me several times, they're parsed the same as `.md` anyway
* Redirects some old links to `CONTRIBUTING.md` to https://typescript-eslint.io/contributing

### CODE REVIEW

File: docs/maintenance/Versioning.mdx [hunk 1]
```diff
@@ -15,11 +15,21 @@ Additionally, we promote to the `latest` tag on NPM once per week, **on Mondays
...
 
 The latest version under the `canary` tag **(latest commit to `main`)** is:
 
-<a href="https://www.npmjs.com/package/@typescript-eslint/parser"><img src="https://img.shields.io/npm/v/@typescript-eslint/parser/canary.svg?style=flat-square" alt="NPM Version" /></a>
+<a href="https://www.npmjs.com/package/@typescript-eslint/parser">
+  <img
+    src="https://img.shields.io/npm/v/@typescript-eslint/parser/canary.svg?style=flat-square"
+    alt="NPM Version"
+  />
+</a>
```
Comment (JoshuaKGoldberg) [inline docs/maintenance/Versioning.mdx:32]: 🤷 I guess Prettier more aggressively formats inline HTML snippets in `.mdx` files than `.md`.

**Comment by Josh-Cena** [comment]
PSA: `.mdx` and `.md` may have different behaviors in Docusaurus v3. We are still figuring it out. For example, you won't be able to use HTML comments in `.mdx` because MDX v2 only allows `{/* */}` style ones. In Docusaurus v3, we do recommend you using `.mdx` if you actually use MDX syntax, though.

**Review by bradzacher — APPROVED** [review]
nice_clap

**Comment by bradzacher** [comment]
fix conflicts and :shipit:

**Pull request merged**